src/main/java/nurseybook/model/task/UniqueTaskList.java

```diff
@@ -138,26 +155,26 @@ public void deleteGhostTasks() {
      * the future task is added as a ghost task to the TaskList.
      */
     public void addPossibleGhostTasksWithMatchingDate(LocalDate keyDate) {
-        List<Task> ghostTaskList = new ArrayList<Task>();
+        List<GhostTask> ghostTaskList = new ArrayList<GhostTask>();
         for (Task task : this.internalList) {
             if (task.checkIfTaskRecurring() && task.checkIfRealTask()) {
-                Task ghostTask = addFutureGhostTasksWithMatchingDate(task, keyDate);
+                GhostTask ghostTask = addFutureGhostTasksWithMatchingDate((RealTask) task, keyDate);
                 if (ghostTask != null) {
                     ghostTaskList.add(ghostTask);
                 }
             }
         }
 
-        for (Task task : ghostTaskList) {
-            this.add(task);
+        for (GhostTask ghostTask : ghostTaskList) {
+            this.add(ghostTask);
         }
     }
 
     /**
      * Checks if any of the given recurring task's future occurrences coincide with the given keyDate. If it does,
      * the future task is added as a ghost task to the TaskList.
      */
-    private Task addFutureGhostTasksWithMatchingDate(Task task, LocalDate keyDate) {
+    private GhostTask addFutureGhostTasksWithMatchingDate(RealTask task, LocalDate keyDate) {
```

Comment:
Perhaps the method name and javadoc can be changed to better match what the method actually does? The addition of the ghost task to the task list is done by the `addPossibleGhostTasksWithMatchingDate` method.

Reply:
Noted! I will change the JavaDocs and method name.

Reply:
Fyi the actual implementation of methods relevant to `ViewSchedule` will be updated in a future PR.